packages/world/src/IHooks.sol

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { EntityId } from "./EntityId.sol";
+import { ProgramId } from "./ProgramId.sol";
+import { Vec3 } from "./Vec3.sol";
+import { ObjectTypeId } from "./ObjectTypeId.sol";
+import { ObjectAmount } from "./ObjectTypeLib.sol";
+
+/**
+ * caller is the entity that called the system and triggered the hook
+ * target is the entity for which the hook is being called
+ */
+interface IHooks {
```

Comment:
small thing but should we call this something like `IProgramHooks`? just to shy away from the built-in `Hooks.sol` lib/table in Store

I know there's no overlap right now but just wary of a potential overlap later

Reply:
ye I agree, I just think IHooks is more comfortable when using it because it is a little bit shorter. Maybe we could call them events? or come up with some other way for calling hooks.

Originally I had directly implemented the hooks in the ProgramId lib so you could do `program.onMine(...)`, but I didn't like that the same hook was defined in the interface (to be able to use encodeCall) and in the library. Also most hooks are only used in a single place, so adding it to the library didn't make much sense. And that meant that the way they were called (safeCall, staticcall, etc) was done in the lib as well so it felt like too much abstraction and less control within the systems that call them.

Reply:
Yeah validateProgram and getDisplayURI don't feel like hooks, so we could also move them to a different interface. I kinda like IProgram, but could be confusing as it sounds like it is something your program has to inherit from?

Reply:
what about `abstract contract Program` and implement the hooks but revert? like https://github.com/latticexyz/mud/blob/main/packages/world/src/Module.sol

Reply:
what is this interface solving for? so you can do `IHooks(program).someHook()`?

Reply:
This also solves the issue with naming, ie hooks can be called ISleepHook.onSleep, and non-hooks can be IProgramValidator.validateProgram, IDisplay.getDisplayURI, etc